src/Mod/Sketcher/Gui/ViewProviderSketch.cpp

```diff
@@ -366,6 +366,8 @@ void ViewProviderSketch::activateHandler(DrawSketchHandler *newHandler)
     sketchHandler->sketchgui = this;
     sketchHandler->activated(this);
 
+    editCoinManager->setConstraintSelectability(false);

```

Comment:
I tend to think that it is not the ViewProvider's call to decide on constraint selectability. Managing the interaction with the rest of the scenograph is a responsibility of DrawSketchHandler (and the corresponding derived class implementing the command). 

Surely DrawSketchHandler can set a default that can be overriden by any derived class that finds it useful to select a constraint. For example, PaddleStroke's contextual constraining might benefit from the ability to edit the value of a datum constraint during execution (for example to correct a previously wrongly introduced one).

If you agree, a private function setConstraintSelectability could be created in ViewProviderSketch (we do not want anyone with a pointer tweaking functionality that under SRP belongs to DrawSketchHandler). Then there is an attorney class to regulate the level of friendship of DrawSketchHandler and ViewProviderSketch (ViewProviderSketchDrawSketchHandlerAttorney) that should implement this. DrawSketchHandler accesses private stuff of ViewProviderSketch via this attorney. I put in place this mechanism to force SRP between VP and DrawSketchHandler. The classes are very much related and should be partially coupled, but as loosest as possible. This attorney is there to kill the tentation of mixing responsibilities.